### PR TITLE
Add e2e test and accompanying documentation

### PR DESCRIPTION
**Pull request by wallrj**
Here's an approach that:
* builds navigator 
* builds navigator Docker images on a Minikube VM
* installs that compiled navigator in Minikube
* tests that it runs and 
* tests that it registers custom resources that can be created and deleted.

I considered trying to re-use the stuff at https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md
But decided that I'd try this simple approach before investing too much time understanding that much larger framework.

Closes https://gitlab.jetstack.net/marshal/navigator/issues/16

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,11 +27,12 @@ GOPATH ?= /tmp/go
...
-	# test - runs go_test target
-	# build	- runs generate, and then go_build targets
-	# generate - generates pkg/client/ files
-	# verify - verifies generated files & scripts
+	# all				- runs verify, build and docker_build targets
+	# test			- runs go_test target
+	# e2e-test	- runs e2e tests
+	# build			- runs generate, and then go_build targets
+	# generate	- generates pkg/client/ files
+	# verify		- verifies generated files & scripts
```
Comment (simonswine) [inline Makefile]: Not too sure the formating looks strange, I guess you need convert tabs to spaces in the comments

**Comment by simonswine** [comment]
This is the error that I get on a fresh minikube:

```
+ kubectl create --namespace navigator-e2e-database1 --filename /home/christian/.golang/packages/src/github.com/jetstack-experimental/navigator/docs/quick-start/es-cluster-demo.yaml
error: unable to recognize "/home/christian/.golang/packages/src/github.com/jetstack-experimental/navigator/docs/quick-start/es-cluster-demo.yaml": no matches for navigator.jetstack.io/, Kind=ElasticsearchCluster
Makefile:46: recipe for target '.hack_e2e' failed
make: *** [.hack_e2e] Error 1
```

File: Makefile (hunk 1 above)
Comment (wallrj) [reply Makefile]: Fixed.

**Comment by wallrj** [comment]
> In general looking good to me, but I think you should define a .dockerignore file (https://docs.docker.com/engine/reference/builder/) and make sure it works the first time with an empty cluster without CRD definintions.

Done.

**Comment by wallrj** [comment]
> This is the error that I get on a fresh minikube:

I haven't seen that error. 
Even with a fresh minikube.

```
minikube version: v0.21.0

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:33:11Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-07-26T00:12:31Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

```

**Comment by wallrj** [comment]
I'll merge this and improve the tests in followup branches.

**Comment by munnerz** [comment]
@simonswine only reason I can think for that error, is the delay in registered a CRD.

Do we retry that command (`kubectl create --namespace navigator-e2e-database1 --filename /home/christian/.golang/packages/src/github.com/jetstack-experimental/navigator/docs/quick-start/es-cluster-demo.yaml`)?

**Comment by wallrj** [comment]
> Do we retry that command

No, but perhaps we should consider removing the EnsureTPR code from the navigator controller and instead instructing users to install the TPRs manually. As discussed yesterday.

**Comment by munnerz** [comment]
@wallrj agreed - but we still need to retry these commands. If I recall correctly, it's very difficult to determine if the API server is ready yet aside from attempting to use it over and over. The discovery API will return that the type is registered before the actual go-restful endpoints are available afaik.

Let's see how the e2e tests run in CI and we'll go from there 😄

**Pull request merged**